crates/extensions/tedge_http_ext/src/actor.rs

```diff
@@ -0,0 +1,111 @@
+use crate::{HttpConfig, HttpError, HttpRequest, HttpResponse};
+use async_trait::async_trait;
+use tedge_actors::{Actor, Address, ChannelError, Mailbox, Message, Recipient, Sender};
+
+pub(crate) struct HttpActor {
+    client: reqwest::Client,
+}
+
+impl HttpActor {
+    pub(crate) fn new(_config: HttpConfig) -> Result<Self, HttpError> {
+        let client = reqwest::Client::builder().build()?;
+        Ok(HttpActor { client })
+    }
+}
+
+#[async_trait]
+impl Actor for HttpActor {
+    type Input = (usize, HttpRequest);
+    type Output = (usize, Result<HttpResponse, HttpError>);
+    type Mailbox = Mailbox<Self::Input>;
+    type Peers = Recipient<Self::Output>;
+
+    async fn run(
+        self,
+        mut requests: Self::Mailbox,
+        mut client: Self::Peers,
+    ) -> Result<(), ChannelError> {
+        while let Some((client_id, request)) = requests.next().await {
+            // Forward the request to the http server
+            let request = request.into();
+
+            // Await for a response
+            let response = match self.client.execute(request).await {
```

Comment:
Is there a particular thing preventing this from running in a separate task? I imagine on some level we want to limit the number of in flight requests at any given time, and this currently will shut down when client is closed.

Reply:
To address this issue, the HTTP actor is now using a [`ConcurrentServiceMessageBox`](https://github.com/thin-edge/thin-edge.io/pull/1622/files#diff-85d9e5c8a7a180c4ba09343d893d980d879c01f58dd376d6414d477a9b626cefR154) to handle requests concurrently.